### PR TITLE
CMake: Add support for cygwin paths when using msvc

### PR DESCRIPTION
**Pull request by dnakamura**
Pass paths through cygpath when generating ddr files, so  that they will
be understood by msvc

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by dnakamura** [comment]
@genie-omr build all

**Comment by dnakamura** [comment]
@genie-omr build all

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: cmake/modules/ddr/GenerateStub.cmake [hunk 1]
```diff
@@ -27,10 +27,28 @@ if(NOT EXISTS "${input_file}")
...
+		# no defined tool to convert path names. Do nothing
+		set(${output} "${filename}")
+	endif()
+endmacro()
 
-if(NOT rc EQUAL 0)
-	# input didn't have any DDR directives, so just dump an empty file
+execute_process(COMMAND grep -lE "@ddr_(namespace|options):"  ${input_file} RESULT_VARIABLE rc)
+if(rc)
+	#input didnt have any ddr directives, so just dump an empty file
```
[the remark this answers is not in the document]
Comment (dnakamura) [reply cmake/modules/ddr/GenerateStub.cmake:51]: Artifact of an automated rebase

**Comment by dnakamura** [comment]
@genie-omr build all

File: cmake/modules/ddr/GenerateStub.cmake [hunk 2]
```diff
@@ -27,8 +27,26 @@ if(NOT EXISTS "${input_file}")
...
 
-execute_process(COMMAND grep -E -q "@ddr_(namespace|options):" ${input_file} RESULT_VARIABLE rc)
+macro(convert_path output filename)
+	if(PATH_TOOL)
+		execute_process(
+			COMMAND ${PATH_TOOL} "${filename}"
+			OUTPUT_VARIABLE _converted_path
+			RESULT_VARIABLE _convert_rc
+		)
+		if(_convert_rc)
```
Comment (keithc-ca) [inline cmake/modules/ddr/GenerateStub.cmake:37]: For consistency, this should say `if(NOT _convert_rc EQUAL 0)` (see lines 50 & 59).

**Comment by youngar** [comment]
@keithc-ca any more comments?

**Review by keithc-ca — APPROVED** [review]
I'm happy with the current state of this. I was trying to fix my Windows build environment to test it out before offering my final opinion, but I don't see a reason to hold this up any longer.

**Comment by youngar** [comment]
@genie-omr build all

**Pull request merged**